endpoints/openrtb2/amp_auction_test.go

```diff
@@ -1297,6 +1275,81 @@ func TestBuildAmpObject(t *testing.T) {
 	}
 }
 
+func TestIdGeneration(t *testing.T) {
+	testCases := []struct {
+		description                  string
+		givenInTagId                 string
+		givenInStoredRequest         json.RawMessage
+		givenGenerateStoredRequestId bool
+		expected                     int
```

Comment:
Might want to call this `expectedIDLength` as at first I was wondering how you arranged it to have generated ID always be "36".